Sources/SentryCrash/Recording/Tools/SentryCrashThread.c

```diff
@@ -53,3 +53,68 @@ sentrycrashthread_getThreadName(const SentryCrashThread thread, char *const buff
     const pthread_t pthread = pthread_from_mach_thread_np((thread_t)thread);
     return pthread_getname_np(pthread, buffer, (unsigned)bufLength) == 0;
 }
+
+bool
+sentrycrashthread_getQueueName(const SentryCrashThread thread, char *const buffer, int bufLength)
+{
+    // WARNING: This implementation is no longer async-safe!
+
+    integer_t infoBuffer[THREAD_IDENTIFIER_INFO_COUNT] = { 0 };
+    thread_info_t info = infoBuffer;
+    mach_msg_type_number_t inOutSize = THREAD_IDENTIFIER_INFO_COUNT;
+    kern_return_t kr = 0;
+
+    kr = thread_info((thread_t)thread, THREAD_IDENTIFIER_INFO, info, &inOutSize);
+    if (kr != KERN_SUCCESS) {
+        SentryCrashLOG_TRACE(
+            "Error getting thread_info with flavor THREAD_IDENTIFIER_INFO from mach thread : %s",
+            mach_error_string(kr));
+        return false;
+    }
+
+    thread_identifier_info_t idInfo = (thread_identifier_info_t)info;
+    if (!sentrycrashmem_isMemoryReadable(idInfo, sizeof(*idInfo))) {
+        SentryCrashLOG_DEBUG("Thread %p has an invalid thread identifier info %p", thread, idInfo);
+        return false;
+    }
+    dispatch_queue_t *dispatch_queue_ptr = (dispatch_queue_t *)idInfo->dispatch_qaddr;
+    if (!sentrycrashmem_isMemoryReadable(dispatch_queue_ptr, sizeof(*dispatch_queue_ptr))) {
+        SentryCrashLOG_DEBUG(
+            "Thread %p has an invalid dispatch queue pointer %p", thread, dispatch_queue_ptr);
+        return false;
+    }
+    // thread_handle shouldn't be 0 also, because
+    // identifier_info->dispatch_qaddr =  identifier_info->thread_handle +
+    // get_dispatchqueue_offset_from_proc(thread->task->bsd_info);
+    if (dispatch_queue_ptr == NULL || idInfo->thread_handle == 0 || *dispatch_queue_ptr == NULL) {
+        SentryCrashLOG_TRACE("This thread doesn't have a dispatch queue attached : %p", thread);
+        return false;
+    }
+
+    dispatch_queue_t dispatch_queue = *dispatch_queue_ptr;
+    const char *queue_name = dispatch_queue_get_label(dispatch_queue);
+    if (queue_name == NULL) {
+        SentryCrashLOG_TRACE("Error while getting dispatch queue name : %p", dispatch_queue);
+        return false;
+    }
+    SentryCrashLOG_TRACE("Dispatch queue name: %s", queue_name);
+    int length = (int)strlen(queue_name);
+
+    // Queue label must be a null terminated string.
+    int iLabel;
+    for (iLabel = 0; iLabel < length + 1; iLabel++) {
+        if (queue_name[iLabel] < ' ' || queue_name[iLabel] > '~') {
+            break;
+        }
+    }
+    if (queue_name[iLabel] != 0) {
+        // Found a non-null, invalid char.
+        SentryCrashLOG_TRACE("Queue label contains invalid chars");
+        return false;
+    }
```

Comment:
^ This is the only new code that has been added here since the original implementation that was crashy (https://github.com/kstenerud/KSCrash/issues/207#issuecomment-402963340)

This won't fix the crash -- I mentioned in my other PR this comment in some WebKit code: https://github.com/WebKit/WebKit/blob/09225dc168d445890bb0e2a5fa8bc19aef8556f2/Source/WebCore/page/cocoa/ResourceUsageThreadCocoa.mm#L119

The unsafe part is line 94 where you dereference the pointer to the dispatch queue, which could have been deallocated by the time you dereference it.  The comment on `searchQueueNames` saying this might crash is still accurate. Do you still want to expose the option in that case?